st2auth/st2auth/app.py

```diff
@@ -63,6 +64,10 @@ def setup_app(config={}):
                      capabilities=capabilities,
                      config_args=config.get('config_args', None))
 
+        # pysaml2 uses subprocess communicate which calls communicate_with_poll
+        if cfg.CONF.auth.sso and cfg.CONF.auth.sso_backend == 'saml2':
+            use_select_poll_workaround(nose_only=False)
```

Comment:
I would have thought it's a pure Python implementation. Just curious, do yoy happen to know when it needs to spawn subprocesses?

Reply:
The pysaml2 module is running xmlsec on a subprocess at https://github.com/IdentityPython/pysaml2/blob/master/src/saml2/sigver.py#L891. This step happens during callback from the Idp and the pysaml2 is verifying/validating the signature of the response.